### PR TITLE
add ssl config options to the plugin

### PR DESCRIPTION
**Pull request by Thoomyy**
Possibility to define the path to the ssl cert for downloading images with file_get_contents
Add the option to switch of the ssl validation for downloading images (should only be used for test purposes)

### CODE REVIEW

**Comment by moduleon** [comment]
Hi thoomyy, thank you for your pull request!
Just a question, why would you need to add the ssl cert for your tests?

**Comment by Thoomyy** [comment]
Hi Moduleon
The SSL cert option is if you have a host, where the cacert didn't set up.
The option to switch of the SSL verification should be only for tests.
If you have a host without a cacert config and you try to download posts with images you will get the error:
"An error occured during social posts array conversion. " More: https://discourse.getgrav.org/t/an-error-occured-during-social-posts-array-conversion/7073 (https://discourse.getgrav.org/t/an-error-occured-during-social-posts-array-conversion/7073)
For me, the error says nothing about the actual problem.
So maybe someone is glad if he has this option to debug.
Regards

Thoomyy

Am Fr., Sept. 21, 2018 06:01 schrieb Kévin Marcachi : Hi thoomyy, thank you for your pull request!
 Just a question, why would you need to add the ssl cert for your tests?

File: src/Manager/PostManager.php [hunk 1]
```diff
@@ -180,7 +180,21 @@ public function storeAttachments(Post $post)
...
+
+        if (isset($config['enablessl']) && $config['enablessl'] == false) {
+        	$arrContextOptions['ssl']['verify_peer'] = 0;
+        	$arrContextOptions['ssl']['verify_peer_name'] = 0;
+        }
+
+        if (isset($config['certpath']) && $config['certpath']) {
+        	$arrContextOptions['ssl']['cafile'] = $config['certpath'];
+        }
+        print_r($arrContextOptions);
```
Comment (moduleon) [inline src/Manager/PostManager.php]: Please, remove this print_r().

File: src/Manager/PostManager.php [hunk 2]
```diff
@@ -180,7 +180,21 @@ public function storeAttachments(Post $post)
      */
     private function downloadFile($url, $basename, $dir)
     {
-        $fileContents = @file_get_contents($url);
+    	$arrContextOptions = [];
```
Comment (moduleon) [inline src/Manager/PostManager.php]: Please add 2 spaces for indentation.

File: src/Manager/PostManager.php [hunk 3]
```diff
@@ -180,7 +180,21 @@ public function storeAttachments(Post $post)
...
+
+    	$config = $this->grav['config']->get('plugins.social-feed');
```
Comment (moduleon) [inline src/Manager/PostManager.php]: Please add 2 spaces for indentation.

File: src/Manager/PostManager.php (hunk 1 above)
Comment (Thoomyy) [reply src/Manager/PostManager.php]: removed

File: src/Manager/PostManager.php (hunk 3 above)
Comment (Thoomyy) [reply src/Manager/PostManager.php]: Hi 
I updated the file indentation. Can you please check that?
https://github.com/Thoomyy/grav-plugin-social-feed/commit/88f05eb7d40269345d3179ed8c8ff1a9e8973223

**Pull request merged**